File: 2/python/solution2.py
limit = 400000

def fibonacci(limit):
    total = 0
    a,b = 1,1
    while True:
        c  = a+b
        if c > limit:
            break
        if c % 2 ==0:
            total += c
        a = b
        b = c 
    return total

total = fibonacci(limit)
print(total)

File: 2/rust/solution2.rs

fn main () {
    let  limit = 400000;
    let  (mut a, mut b) = (1,1);
    let mut total = 0;
    loop {
        let c = a+b;
        if c > limit {
            break;
        }
        if c % 2 == 0 {
            total +=c;
        }
        a = b;
        b = c;
    }    
    println!("{}",total);

}
